client/signup/config/steps-pure.js

```diff
@@ -324,7 +324,7 @@ export function generateSteps( {
 
 		'domain-only': {
 			stepName: 'domain-only',
-			providesDependencies: [ 'siteId', 'siteSlug', 'domainItem' ],
+			providesDependencies: [ 'siteId', 'siteSlug', 'domainItem', 'shouldHideFreePlan' ],
```

Comment:
I don't think we should force `shouldHideFreePlan ` as a `providesDependencies` in every version of the domain step. Take the above line as example, in the domain-only flow it does not look relevant to send a `shouldHideFreePlan` dependency when we don't even have a plans step if the user just wants to buy a domain. This is not a bug, but it could just be confusing to developers in the future if someone wants to add a domain step and wondering why that step should forced to send a prop related to plans.

Going back to [this line](shouldHideFreePlan) which was removed in this PR, every new flow in the future that includes a domain step will be forced to add `shouldHideFreePlan` as a dependency. A problem is that this tightly couples the order of steps and assumes that the plans step will either always be present in the flow, or will always appear _after_ the domains step. What if we have a new flow that shows the plans step _before_  the domains step?(example, check the A/B test flow [onboarding-plan-first](https://github.com/Automattic/wp-calypso/blob/master/client/signup/config/flows-pure.js#L125). Then `shouldHideFreePlan` as a provided dependency in the domain step doesn't look right if the plan step was already completed by the time the domain step loaded in the hypothetical flow. Again, not a bug but it just logically doesn't fit. 

I think it's probably better to keep the existing pattern of adding `shouldHideFreePlan` as an optional dependency([check here how the domain step does it](https://github.com/Automattic/wp-calypso/blob/master/client/signup/config/steps-pure.js#L318)), and retain the lines that were [removed here](https://github.com/Automattic/wp-calypso/pull/45618/files#diff-df013e2a4c6a6c2b5973d50e326984f8L258). We can then completely remove `shouldHideFreePlan` as a dependency for "domain-only" and "select-domain" steps.

Reply:
@niranjan-uma-shankar Can you please fix the link at "this line" in the following sentence. It seems to be incorrect.
> Going back to this line which was removed in this PR, every new flow in the future that includes a domain step will be forced to add shouldHideFreePlan as a dependency.

Reply:
Done

Reply:
I still think we do not need to bring back the code removed [here](https://github.com/Automattic/wp-calypso/pull/45618/files#diff-df013e2a4c6a6c2b5973d50e326984f8L258), since there is anyway a default parameter on the function and we will fallback to it being false whenever there is no value provided. 

However I will move the dependency to optional dependencies so that we don't make this step force to expect that parameter.

Reply:
Fixed at e2bc89cefa6b22738452a6a0eabd8d60e2e8b0af